Expensify/App
# Add tag muted-text BaseHTMLEngineProvider to render html errors

### Details

The new error text:

>You have already set up this bank account on Expensify. If you'd like to add it again, please delete it [here](https://www.expensify.com/settings?param=%7B%22section%22:%22payments%22%7D).

Comes from `Web-Secure` backend and contains a link, so I'll be sending it using HTML. See related [Web-Secure PR](https://github.com/Expensify/Web-Secure/pull/2055)

Added new tag `muted-text` in BaseHTMLEngineProvider, this tag is only for setting the style `mutedTextLabel` when we find the tag. The idea is that if we receive an html error from the back, we wrap it in `<muted-text>` and render it using `RenderHTML`.

Also, we currently render all `<a>` with external urls in `href` as `AnchorForCommentsOnly`. I introduced a custom tag `<comment>` to wrap comment's html, and use it to detect if we are rendering a `<a>` within a comment or not. If we are rendering a `<a>` outside of a comment, like the error coming from the back end, we render it using a `<Text>` because `AnchorForCommentsOnly`:
- Adds a context menu that is not present in links in other parts of the app
- Doesn't align well with the rest of the text in Android



### Fixed Issues

$ https://github.com/Expensify/App/issues/5626
https://github.com/Expensify/Expensify/issues/177017

### Tests / QA

Dev: make sure you have the changes from [this PR](https://github.com/Expensify/Web-Secure/pull/2055) in you workspace

1. Create a new account in OldDot, verify the email
2. Still in OldDot, add an open VBA using data from this [SO (3)](https://stackoverflow.com/c/expensify/a/525/419). Add VBA can be found in: `Settings > Account > Payments > Add Verified Bank Account`
3. Log in with the same account in NewDot
4. Create a workspace and start the add VBA process
5. Choose to add manually the bank account, input `Routing Number: 011401533` and `Account number: 1111222233331111`
6. Click "Save & continue"
7. You should see the following error above the "Save & Continue" button:
![Screen Shot 2021-09-27 at 5 25 33 PM](https://user-images.githubusercontent.com/87341702/135002779-93294b72-cd48-4e38-99ed-287d9a5cade3.png)


### QA Steps


### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
